bigquery_storage/noxfile.py

```diff
@@ -111,13 +111,16 @@ def system(session):
     # Use pre-release gRPC for system tests.
     session.install("--pre", "grpcio")
 
+    session.install("protobuf")
```

Comment:
Why was this installation necessary? Do we need to update `setup.py`?

Reply:
Needed for the `Timestamp` protobuf message used in the snapshot test (included in the `TableModifiers` for the read session).

The `create_read_session`'s docstring says that the `table_modifiers` argument can also be a dict with the same structure - is there a canonical way of providing a valid non-protobuf timestamp, circumventing the type `google.protobuf.Timestamp`?

If not, we should probably indeed add `protobuf` as a dependency so that users can configure `table_modifiers` without import errors and manually installing additional packages.

Reply:
> The `create_read_session`'s docstring says that the table_modifiers argument can also be a dict with the same structure

That doesn't actually work, but if it did the official JSON representation of timestamps is an ISO 8601 string. See: https://github.com/googleapis/gapic-generator/issues/2530 :-(

Reply:
> If not, we should probably indeed add protobuf as a dependency so that users can configure table_modifiers without import errors and manually installing additional packages.

Yes, let's add a protobuf dependency. It will need to match with the required versions for grpc.

Reply:
Thanks for confirming it. I also had trouble initially when using a dict (type mismatch errors, etc.), and found no quick workaround apart from providing a protobuf Timestamp.

Reply:
FWIW, `protobuf` is already a sub-dependency (through `api-core`), and it appears that everything already works without explicitly installing it (kind of expected).

It was probably something on my machine that initially caused an `ImportError`, and cannot reproduce it anymore.